src/test/java/com/github/alexcojocaru/mojo/elasticsearch/StartElasticsearchDataMojoTest.java

```diff
@@ -60,11 +71,68 @@ public void testMojoExecution() throws Exception
         assertNotNull(mojo);
         mojo.execute();
 
-        HttpClient client = HttpClientBuilder.create().build();
-        HttpGet get = new HttpGet("http://localhost:" + mojo.getNode().getHttpPort());
-        HttpResponse response = client.execute(get);
+        HttpGet get = new HttpGet(getUri());
+        HttpResponse response = httpClient.execute(get);
         assertEquals(200, response.getStatusLine().getStatusCode());
         assertEquals(true, mojo.autoCreateIndex);
     }
 
+    private String getUri() throws Exception
+    {
+        return "http://localhost:" + mojo.getNode().getHttpPort();
+    }
+
+    public void testKeepData() throws Exception
+    {
+        assertNotNull(mojo);
+
+        mojo.execute();
+
+        final String indexName = "myindex";
+
+        // succeeds
+        StatusLine statusLine = createIndex(indexName);
+        assertEquals(statusLine.getReasonPhrase(), 2, statusLine.getStatusCode() / 100);
+
+        // fails because exists
+        Thread.sleep(500);
+        statusLine = createIndex(indexName);
+        assertEquals(statusLine.getReasonPhrase(), 4, statusLine.getStatusCode() / 100);
+
+        stopNode();
+        setUp();
+        Thread.sleep(500);
+        mojo.execute();
+
+        // succeeds because deleted and newly created
+        statusLine = createIndex(indexName);
+        assertEquals(statusLine.getReasonPhrase(), 2, statusLine.getStatusCode() / 100);
+
+        stopNode();
+        setUp();
+        Thread.sleep(500);
+        mojo.keepData = true;
+        mojo.execute();
+
+        // fails because existed from previous
+        statusLine = createIndex(indexName);
+        assertEquals(statusLine.getReasonPhrase(), 4, statusLine.getStatusCode() / 100);
+
+    }
+
+    private StatusLine createIndex(final String indexName) throws Exception
+    {
+        final String indexUri = getUri() + "/" + indexName;
+        HttpPut put = new HttpPut(indexUri);
+        HttpResponse response = httpClient.execute(put);
+        final StatusLine statusLine = response.getStatusLine();
+        if (statusLine.getStatusCode() >= 300) {
+            final ByteArrayOutputStream ostream = new ByteArrayOutputStream();
+            response.getEntity().writeTo(ostream);
+            System.out.println(statusLine.toString());
+            System.out.println(ostream.toString());
```

Comment:
If you want to print out the status code and response body, I suggest doing so on success and on error, not just on error. Error is expected sometimes by the test above; getting success from the server in that case would be unexpected and the test should fail.
Also, it would be nice to provide some context along with the status code and the response body; eg. System.out.println("Sent PUT request to " + indexUri + "; received status code '" + statusLine.toString() + "' and message:" + System.getProperty("line.separator") + ostream.toString())